instrumentation/pg/lib/opentelemetry/instrumentation/pg/patches/connection.rb

```diff
@@ -123,43 +123,56 @@ def generated_postgres_regex
             @generated_postgres_regex ||= Regexp.union(PG::Constants::POSTGRES_COMPONENTS.map { |component| PG::Constants::COMPONENTS_REGEX_MAP[component] })
           end
 
-          def database_name
-            conninfo_hash[:dbname]&.to_s
-          end
-
-          def first_in_list(item)
-            return unless item
-
-            if (idx = item.index(','))
-              item[0...idx]
-            else
-              item
-            end
-          end
-
           def client_attributes
             attributes = {
               'db.system' => 'postgresql',
-              'db.user' => conninfo_hash[:user]&.to_s,
-              'db.name' => database_name,
-              'net.peer.name' => first_in_list(conninfo_hash[:host]&.to_s)
+              'db.user' => user,
+              'db.name' => db
             }
             attributes['peer.service'] = config[:peer_service] if config[:peer_service]
 
             attributes.merge(transport_attrs).reject { |_, v| v.nil? }
           end
 
+          def transport_addr
+            # The hostaddr method is available when the gem is built against
+            # a recent version of libpq.
+            return hostaddr if defined?(hostaddr)
```

Comment:
Huh. I actually didn't know you could use `defined?` in that way (I thought this was maybe calling `defined?(result-of-unintentional-hostaddr-call)` which wouldn't be a good thing to rely on). But wouldn't you know it, it's right there in the docs 🤦 

https://ruby-doc.org/docs/keywords/1.9/Object.html#method-i-defined-3F

Thank you, I learned something new today. 😄

Reply:
I learned it during this PR as well. I saw it as part of module prepending: `super if defined? super`

In this case, we can use `respond_to?`. I wonder if one is definitely faster? 🤷🏻

Reply:
I mean; either is probably fine. But my first meeting of the morning cancelled so that means its time for [MicroBenchmark, Thanksgiving Edition 🦃](https://gist.github.com/ahayworth/bdd5f12c5ce08a075289bfa4530a468c)

Looks like `defined?` may be a tiny bit faster? It probably doesn't have to go through things like `method_missing` and whatever. Here are the results:
```
/-------------------------------------------------------------\
|                  user     system      total        real     |
|    defined?:   0.094393   0.000956   0.095349 (  0.095374)  |
| respond_to?:   0.099900   0.001003   0.100903 (  0.100944)  |
\-------------------------------------------------------------/
   \
    \                                  ,+*^^*+___+++_
     \                           ,*^^^^              )
      \                       _+*                     ^**+_
       \                    +^       _ _++*+_+++_,         )
                _+^^*+_    (     ,+*^ ^          \+_        )
               {       )  (    ,(    ,_+--+--,      ^)      ^\
              { (@)    } f   ,(  ,+-^ __*_*_  ^^\_   ^\       )
             {:;-/    (_+*-+^^^^^+*+*<_ _++_)_    )    )      /
            ( /  (    (        ,___    ^*+_+* )   <    <      \
             U _/     )    *--<  ) ^\-----++__)   )    )       )
              (      )  _(^)^^))  )  )\^^^^^))^*+/    /       /
            (      /  (_))_^)) )  )  ))^^^^^))^^^)__/     +^^
           (     ,/    (^))^))  )  ) ))^^^^^^^))^^)       _)
            *+__+*       (_))^)  ) ) ))^^^^^^))^^^^^)____*^
            \             \_)^)_)) ))^^^^^^^^^^))^^^^)
             (_             ^\__^^^^^^^^^^^^))^^^^^^^)
               ^\___            ^\__^^^^^^))^^^^^^^^)\\
                    ^^^^^\uuu/^^\uuu/^^^^\^\^\^\^\^\^\^\
                       ___) >____) >___   ^\_\_\_\_\_\_\)
                      ^^^//\\_^^//\\_^       ^(\_\_\_\)
                        ^^^ ^^ ^^^ ^
```